src/wp-includes/l10n/class-wp-translation-controller.php

```diff
@@ -0,0 +1,420 @@
+	 *
+	 * @since 6.5.0
+	 * @var array<string, array<string, array<string, WP_Translation_File|false>>>
+	 */
+	protected $loaded_files = array();
+
+	/**
+	 * Returns the WP_Translation_Controller singleton.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return WP_Translation_Controller
+	 */
+	public static function instance(): WP_Translation_Controller {
+		static $instance;
+
+		if ( ! $instance ) {
+			$instance = new self();
+		}
+
+		return $instance;
+	}
+
+	/**
+	 * Returns the current locale.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return string Locale.
+	 */
+	public function get_locale(): string {
+		return $this->current_locale;
+	}
+
+	/**
+	 * Sets the current locale.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $locale Locale.
+	 */
+	public function set_locale( string $locale ) {
+		$this->current_locale = $locale;
+	}
+
+	/**
+	 * Loads a translation file for a given text domain.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $translation_file Translation file.
+	 * @param string $textdomain       Optional. Text domain. Default 'default'.
+	 * @param string $locale           Optional. Locale. Default current locale.
+	 * @return bool True on success, false otherwise.
+	 */
+	public function load_file( string $translation_file, string $textdomain = 'default', string $locale = null ): bool {
+		if ( null === $locale ) {
+			$locale = $this->current_locale;
+		}
+
+		$translation_file = realpath( $translation_file );
+
+		if ( false === $translation_file ) {
+			return false;
+		}
+
+		if (
+			isset( $this->loaded_files[ $translation_file ][ $locale ][ $textdomain ] ) &&
+			false !== $this->loaded_files[ $translation_file ][ $locale ][ $textdomain ]
+		) {
+			return null === $this->loaded_files[ $translation_file ][ $locale ][ $textdomain ]->error();
+		}
+
+		if (
+			isset( $this->loaded_files[ $translation_file ][ $locale ] ) &&
+			array() !== $this->loaded_files[ $translation_file ][ $locale ]
+		) {
+			$moe = reset( $this->loaded_files[ $translation_file ][ $locale ] );
+		} else {
+			$moe = WP_Translation_File::create( $translation_file );
+			if ( false === $moe || null !== $moe->error() ) {
+				$moe = false;
+			}
+		}
+
+		$this->loaded_files[ $translation_file ][ $locale ][ $textdomain ] = $moe;
+
+		if ( ! $moe instanceof WP_Translation_File ) {
+			return false;
+		}
+
+		if ( ! isset( $this->loaded_translations[ $locale ][ $textdomain ] ) ) {
+			$this->loaded_translations[ $locale ][ $textdomain ] = array();
+		}
+
+		$this->loaded_translations[ $locale ][ $textdomain ][] = $moe;
+
+		return true;
+	}
+
+	/**
+	 * Unloads a translation file for a given text domain.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Translation_File|string $file       Optional. Translation file instance or file name. Default all files.
+	 * @param string                     $textdomain Optional. Text domain. Default 'default'.
+	 * @param string                     $locale     Optional. Locale. Default all locales.
```

Comment:
```suggestion
	 * @param WP_Translation_File|string $file       Optional. Translation file instance or file name. Defaults to all files.
	 * @param string                     $textdomain Optional. Text domain. Default 'default'.
	 * @param string                     $locale     Optional. Locale. Defaults to all locales.
```
Unless it's an actual default value, e.g. a specific string, I think we use the `Defaults to ...` wording more often, e.g. `Defaults to the current post`.